### ISSUE TITLE
Cannot create a new page

### ISSUE THREAD

**Issue by ADefWebserver** [oqtane/oqtane.framework#170]
Create new site
Log in as host
Try to add a new page to the site
Get the following error:

> [2019-11-12T02:51:47.988Z] Error: System.Text.Json.JsonException: A possible object cycle was detected which is not supported. This can either be due to a cycle or if the object depth is larger than the maximum allowed depth of 0.
>    at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_SerializerCycleDetected(Int32 maxDepth)
>    at System.Text.Json.JsonSerializer.Write(Utf8JsonWriter writer, Int32 originalWriterDepth, Int32 flushThreshold, JsonSerializerOptions options, WriteStack& state)
>    at System.Text.Json.JsonSerializer.WriteCore(Utf8JsonWriter writer, PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)
>    at System.Text.Json.JsonSerializer.WriteCore(PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)
>    at System.Text.Json.JsonSerializer.WriteCoreString(Object value, Type type, JsonSerializerOptions options)
>    at System.Text.Json.JsonSerializer.Serialize[TValue](TValue value, JsonSerializerOptions options)
>    at Oqtane.Services.LogService.Log(Nullable`1 PageId, Nullable`1 ModuleId, Nullable`1 UserId, String category, String feature, LogFunction function, LogLevel level, Exception exception, String message, Object[] args) in C:\Users\Michael\Source\Repos\ADefWebserver\oqtane.framework\Oqtane.Client\Services\LogService.cs:line 54
>    at Oqtane.Modules.ModuleBase.Log(LogLevel level, Exception exception, String message, Object[] args) in C:\Users\Michael\Source\Repos\ADefWebserver\oqtane.framework\Oqtane.Client\Modules\ModuleBase.cs:line 160
>    at Oqtane.Modules.ModuleBase.Logger.LogError(Exception exception, String message, Object[] args) in C:\Users\Michael\Source\Repos\ADefWebserver\oqtane.framework\Oqtane.Client\Modules\ModuleBase.cs:line 219
>    at Oqtane.Modules.Admin.Pages.Add.SavePage() in C:\Users\Michael\Source\Repos\ADefWebserver\oqtane.framework\Oqtane.Client\Modules\Admin\Pages\Add.razor:line 323
>    at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle)

**Comment by abukres** [comment]
I get this error when I click on the Save button. This is new repo clone.

 Error: There was an unhandled exception on the current circuit, so this circuit will be terminated. For more details turn on detailed exceptions in 'CircuitOptions.DetailedErrors'.

**Comment by ADefWebserver** [comment]
The reason I have the full error is that in the startup.cs file I changed:

            services.AddServerSideBlazor();

To :

            services.AddServerSideBlazor()
                .AddCircuitOptions(options => { options.DetailedErrors = true; });

**Comment by sbwalker** [comment]
The error is being thrown from the Logger when it attempts to serialize an object as part of the structured logging. I think the solution is to wrap the serialization operation in a try/catch.

**Comment by ADefWebserver** [comment]
> I think the solution is to wrap the serialization operation in a try/catch.

Or use Jason.net?

**Comment by sbwalker** [comment]
I would prefer to minimize third party dependencies. In .Net Core 3 it seems that the new System.Text.Json API is the recommended approach.

**Comment by ADefWebserver** [comment]
> I would prefer to minimize third party dependencies. In .Net Core 3 it seems that the new System.Text.Json API is the recommended approach.

Agreed.

**Comment by sbwalker** [comment]
So this ended up being a completely different problem than what the error message suggested. I had refactored the PageController so that a non-admin is able to create pages and changed the auth attribute to:

        [Authorize(Roles = Constants.RegisteredRole)]

It turns out that the default "host" account was not a member of the Registered Users role. So I had to add some logic in the a ClaimsPrincipalFactory to add the host user to this role by default.

The serialization issue is another unrelated issue and I had run into this in the server logging methods previously and resolved by using the native Exception.ToString() rather than attempting to serialize the object.

I will check in these changes later today.
